src/fontman/legacy_font.h

```diff
@@ -0,0 +1,135 @@
+
+    virtual ~LegacyFont() override;
+
+    void loadFont(int fontId);
+
+    /*!
+     * \brief Measure the size of the multiline text block in pixels
+     * \param text Multi-line text string
+     * \param text_size The byte size of the text string to print
+     * \param max_line_length maximum length of the line in characters
+     * \param cut Cut the part of string after the line length and put the "..." to the end (Note: input string will remina unmodified)
+     * \param fontSize The size of the TTF font glyph (unused for raster fonts)
+     * \return Width and height of the text block in pixels
+     */
+    PGE_Size textSize(const char* text, size_t text_size,
+                      uint32_t max_line_lenght = 0,
+                      bool  cut = false, uint32_t fontSize = 14) override;
+
+    /*!
+     * \brief Get a size of one glyph
+     * \param utf8char Pointer to one UTF-8 character
+     * \param charNum Character number in the line (required to compute the size of the taboluation)
+     * \param fontSize Size of TTF font glyph
+     * \return Width and height of one glyph in pixels
+     */
+    PGE_Size glyphSize(const char *utf8char, uint32_t charNum = 0, uint32_t fontSize = 14) override;
+
+    /*!
+     * \brief Print the multiline text block on the screen
+     * \param text Multi-line text string
+     * \param text_size The byte size of the text string to print
+     * \param x Hotizontal screen position (at left-top corner of the block)
+     * \param y Vertical screen position (at left-top corner of the block)
+     * \param Red Red channel colour
+     * \param Green Green channel colour
+     * \param Blue Blue channel colour
+     * \param Alpha Alpha channel level
+     * \param fontSize The size of the TTF font glyph (unused for raster fonts)
+     */
+    void printText(const char* text, size_t text_size,
+                   int32_t x, int32_t y,
+                   float Red=1.f, float Green=1.f, float Blue=1.f, float Alpha=1.f,
+                   uint32_t fontSize = 0) override;
+
+    bool isLoaded() const override;
+
+    std::string getFontName() const override;
+    FontType getFontType() const override;
+
+private:
+    //! font is fine
+    bool m_isReady = false;
+
+    //! Handalable name of the font
+    std::string m_fontName;
+
+    //! Bank of legacy font textures
+    StdPicture *m_textures[10];
+
+    /**
+     * @brief Raster font glyph
+     */
+    struct RasChar
+    {
+        uint32_t texId = 0; //!< Texture index in m_textures array
+        uint32_t tX = 0; //!< Horizontal pixel offset on texture
+        uint32_t tY = 0; //!< Vertical pixel offset on texture
+        uint32_t tW = 0; //!< Pixels width of the texture fragment
+        uint32_t tH = 0; //!< Pixels height of the texture fragment
+        uint32_t width = 0; //!< Width of glyph
+        bool valid = false; //!< Is a valid glyph
+        int32_t x = 0;//!< X pixel offset
+        int32_t y = 0;//!< Y pixel offset
```

Comment:
Can be shrunk a lot (could be nice to do this at RasterFont as well).

```
struct RasChar
{
    uint8_t texId = 0;
    bool valid = false;
    uint16_t tX = 0;
    uint16_t tY = 0;
    uint8_t tW = 0;
    uint8_t tH = 0;
    uint8_t width = 0;
    int8_t x = 0;
    int8_t y = 0;
};
```

Size: 36 bytes -> 12 bytes

Reply:
I guess, I also can get rid of x and y, they are actually unused.

Reply:
Done just now, please take a look.